server/src/main/java/org/elasticsearch/plugins/internal/DocumentParsingProvider.java

```diff
@@ -13,35 +13,27 @@
  */
 public interface DocumentParsingProvider {
```

Comment:
That's ok with me 👍  I couldn't find a much better name right now, but `DocumentParsingProvider` is a bit strange, it makes this seem like a `DocumentParserProvider`. 
This provider, however, provides two components: one observes parsing to track the size of a document, the other is a more of action / success listener invoked when indexing is done. While the first one is somehow parsing related, the 2nd is not at all - it just happens to see data that was parsed at some point.
But we can always come back to this if we figure out better names. Let's move on 👍

Reply:
this is good observation, and in fact we will be adding a 3rd Accumulator. 
this will probably make the name even worse. let's try to rename this after the whole infrastructure is merge in a separate PR as a rename of this class will pollute heavily

Reply:
I agree with you both :) 
It would be great to come up with a better name (or separate interfaces if a good name cannot be found, it's usually a tell-tale). But better to do later when we have the whole picture, and in a separate, "mechanical" PR.